main.py

```diff
@@ -7,6 +7,9 @@
 from src.algos.cbet import train as train_cbet
 from src.algos.ride import train as train_ride
 
+import torch.multiprocessing
+torch.multiprocessing.set_sharing_strategy('file_system')
+
```

Comment:
Workaround for torch bug

Reply:
What was the error you were getting?
Is there any 'side effect' of doing this?
Why not doing it where `mp` is already imported? ([Here](https://github.com/sparisi/cbet/blob/cdeb335b0a869c71278aaf74f1128d128b58e151/src/init_models_and_states.py#L17))

Reply:
Here is the error I was getting, including a description:
```
(cbet_python) nroth@nroth-pc:~/cbet$ OMP_NUM_THREADS=1 python main.py --model cbet --env HabitatNav-hotel_0 --checkpo
int=/media/nroth/Data/cbet/habitat/21-10-08/baselines-01-26-39-898316/eHabitatNav-apartment_0-ri1-na4-nb80-ul100-nt4-
bs32-hb128-d0.99-crp0.001-lr0.0001-ec0.0005-irc0.0-tf5000000-si20-c-mcbet/model.tar --intrinsic_reward_coef=0.005
Creating log directory: logs/cbet-20220108-150827
Saving arguments to logs/cbet-20220108-150827/meta.json
Saving messages to logs/cbet-20220108-150827/out.log
Saving logs data to logs/cbet-20220108-150827/logs.csv
Saving logs' fields to logs/cbet-20220108-150827/fields.csv
[INFO:35967 init_models_and_states:44 2022-01-08 15:08:27,577] Using CUDA.
2022-01-08 15:08:27,596 Initializing dataset PointNav-v1
2022-01-08 15:08:27,597 initializing sim Sim-v0
2022-01-08 15:08:28,781 Initializing task Nav-v0
loading model from /media/nroth/Data/cbet/habitat/21-10-08/baselines-01-26-39-898316/eHabitatNav-apartment_0-ri1-na4-nb80-ul100-nt4-bs32-hb128-d0.99-crp0.001-lr0.0001-ec0.0005-irc0.0-tf5000000-si20-c-mcbet/model.tar
Traceback (most recent call last):
  File "/home/nroth/cbet/main.py", line 29, in <module>
  File "/home/nroth/cbet/main.py", line 22, in main
  File "/home/nroth/cbet/src/algos/cbet.py", line 155, in train
  File "/home/nroth/cbet/src/init_models_and_states.py", line 140, in init_models_and_states
  File "/home/nroth/cbet/src/utils.py", line 132, in create_buffers
  File "/home/nroth/cbet/cbet_python/lib/python3.9/site-packages/torch/_tensor.py", line 426, in share_memory_
  File "/home/nroth/cbet/cbet_python/lib/python3.9/site-packages/torch/storage.py", line 145, in share_memory_
RuntimeError: falseINTERNAL ASSERT FAILED at "../aten/src/ATen/MapAllocator.cpp":263, please report a bug to PyTorch. unable to open shared memory object </torch_35967_985> in read-write mode
```
Here's a video watching nvtop and my shared memory handles -- as you can see, there is plenty of system and GPU memory, as well as SHM handles, from start until crash: [bug.mov.zip](https://github.com/sparisi/cbet/files/7834026/bug.mov.zip)

And here, you can see that limits are configured reasonably:
```
$ ipcs -l

------ Messages Limits --------
max queues system wide = 32000
max size of message (bytes) = 8192
default max size of queue (bytes) = 16384

------ Shared Memory Limits --------
max number of segments = 4096
max seg size (kbytes) = 18014398509465599
max total shared memory (kbytes) = 18014398509481980
min seg size (bytes) = 1

------ Semaphore Limits --------
max number of arrays = 32000
max semaphores per array = 32000
max semaphores system wide = 1024000000
max ops per semop call = 500
semaphore max value = 32767
```

Reply:
Is there any side effect of doing this?

You can read more about the difference between the default ("file_descriptor") and this method ("file_system") here: https://pytorch.org/docs/stable/multiprocessing.html#sharing-strategies

Basically, the default method does not require an extra daemon to sleep-wait for the process group to die and clean up SHM handles. I switched because it seems like PyTorch was not using shared memory correctly, and that was causing a crash.

Reply:
**Why not doing it where mp is already imported?**

Great idea! I moved the fix to that location, and training and testing are still verified-working.